### ISSUE TITLE
Missing autoload on `dired-jump`

### ISSUE THREAD

**Issue by darkfeline** [syl20bnr/spacemacs#5550]
`SPC f j` `dired-jump` is missing an autoload.  `dired` must be loaded some other way before that binding/command will work.

On 744cd0f

**Comment by StreakyCobra** [comment]
Is it? https://github.com/emacs-mirror/emacs/blob/5283593ced29a2c98e4af3607a68ee7a6d10fcac/lisp/dired-x.el#L408

**Comment by ikhthiandor** [comment]
I'm getting "wrong type argument commandp" if I invoke deer or ranger in dired-mode. Is it related to this issue? 

Here is my system info.
#### System Info
- OS: gnu/linux
- Emacs: 25.1.50.1
- Spacemacs: 0.105.14
- Spacemacs branch: develop (rev. f801cee)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(spacemacs-helm auto-completion better-defaults emacs-lisp
                (version-control :variables version-control-global-margin t version-control-diff-tool 'diff-hl)
                git github org
                (shell :variables shell-default-height 55 shell-default-position 'bottom shell-default-shell 'multi-term)
                (syntax-checking :variables syntax-checking-enable-by-default t)
                html javascript typescript react clojure elm rust
                (go :variables gofmt-command "goimports")
                haskell scala python ipython-notebook django hy-spacemacs ruby php racket
                (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode c-c++-enable-clang-support t)
                asm java yaml semantic vimscript cscope gtags markdown
                (org :variables org-enable-github-support t)
                dash fasd
                (ranger :variables ranger-show-dotfiles t ranger-parent-depth 1 ranger-cleanup-eagerly t ranger-ignored-extensions
                        '("mkv" "iso" "mp4")
                        ranger-override-dired t ranger-max-preview-size 10)
                erc
                (ibuffer :variables ibuffer-group-buffers-by 'projects)
                eyebrowse spacemacs-layouts restclient asciidoc search-engine deft command-log
                (spell-checking :variables spell-checking-enable-by-default nil)
                speed-reading
                (wakatime :variables wakatime-api-key "ea9d7976-7d56-4bba-bda8-61164775857f" wakatime-python-bin "/usr/bin/python3" wakatime-cli-path "/usr/bin/wakatime")
                sql prodigy imenu-list electric-operator electric-case lunaryorn-javascript jb-replace-text jsfmt blog-admin red rtags fancy-narrower)
```

**Comment by darkfeline** [comment]
@StreakyCobra do we not have to register `dired-x`'s autoloads manually?

**Comment by StreakyCobra** [comment]
Ok, so… PR welcome :-)

**Comment by darkfeline** [comment]
I see `vinegar` already loads `dired-x`.  I guess we should remove this from `vinegar` and add a proper `autoload` to... `spacemacs-base`?

**Comment by StreakyCobra** [comment]
`dired` being shipped with emacs, I suppose it's could be set in spacemacs-base yes.

**Comment by darkfeline** [comment]
https://github.com/syl20bnr/spacemacs/pull/5597

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
